src/hooks/useMergeRef.ts

```diff
@@ -0,0 +1,34 @@
+import { ForwardedRef, MutableRefObject, RefObject, useEffect, useRef } from "react";
+
+/*
+ * Example usage:
+ * const Component = React.forwardRef((props, ref) => {
+ *   const internalRef = React.useRef();
+ *   const mergedRef = useMergeRef(ref, internalRef);
+ *   return <div {...props} ref={mergedRef} />;
+ * });
+ */
+
+/**
+ * Returns a single ref callback that merges multiple ref callbacks - replacement for `useMergeRefs` hook
+ * @param refs
+ */
+function useMergeRef<T>(...refs: (RefObject<T> | ForwardedRef<T> | null)[]): RefObject<T> {
```

Comment:
Returns `RefObject`, which can be forwarded later on normally and not a weird function as `useMergeRefs` does